### DISCUSSION TITLE
Tok2vec, OOV and static vectors...again

### DISCUSSION THREAD

**Discussion by python3Berg** [explosion/spaCy#12696] [Help: Model Advice]
I have long struggled with some of the subtleties of vectors, and from other discussions, suspect I am not alone. I am training several models on a very specific business domain that has a very large number of terms that I do not see in the en_core_web_lg vocab. I would love some clarity on best approaches to improved model results.

- I am currently training NER and tok2vec without static vectors. Does the lack of static vectors open up the model to calculate dynamic vectors for all of the meaningful words or is there still a base vocabulary that limits their representation?
- If I were to train my own domain specific vectors, how much might this help vs dynamic tok2vec? I like the dynamic because the deployment size is so much smaller, but I'd rather work around that for more accuracy. If there is a vocabulary constraint, how do I define my own vocabulary if not by the vectors themselves.

I know similar questions have been asked and answered, but I still don't get it. Thanks in advance for the help.

**Comment by rmitsch** [comment]
There are always context-sensitive vectors for all tokens with `tok2vec`. The encode step either uses lexical features or lexical features + static word vectors (where currently you get a 0-vector for tokens without static vectors). So static word vectors don't replace anything, they're just concatenated in addition to the lexical features.

Domain-specific vectors are usually a good idea, if you can get them/train them.
